dataclasses_json/core.py

```diff
@@ -234,9 +234,18 @@ def _decode_generic(type_, value, infer_missing):
     if value is None:
         res = value
     elif _issubclass_safe(type_, Enum):
-        # Convert to an Enum using the type as a constructor.
-        # Assumes a direct match is found.
-        res = type_(value)
+        # we got the enum value
+        for enum_member in type_:
+            # We rely on the user that the enum values are unique
```

Comment:
I don't believe this is necessarily true -- enum names are not necessarily one-to-one with values. There is a decorator `@unique` that enforces it, but the standard library does allow for it